test/units/modules/remote_management/oneview/test_oneview_storage_system.py

```diff
@@ -0,0 +1,300 @@
+    )
+
+
+def test_should_fail_with_missing_required_attributes(resource, mock_ansible_module):
+    mock_ansible_module.params = {
+        'state': 'present',
+        'config': 'config',
+        'data': {
+            'field': 'invalid'
+        }
+    }
+
+    StorageSystemModule().run()
+
+    mock_ansible_module.fail_json.assert_called_once_with(exception=mock.ANY, msg=StorageSystemModule.MSG_MANDATORY_FIELDS_MISSING)
+
+
+def test_should_fail_when_credentials_attribute_is_missing(resource, mock_ansible_module):
+    resource.get_by_name.return_value = []
+
+    mock_ansible_module.params = yaml.load(YAML_STORAGE_SYSTEM_BY_NAME)
+
+    StorageSystemModule().run()
+
+    mock_ansible_module.fail_json.assert_called_once_with(exception=mock.ANY, msg=StorageSystemModule.MSG_CREDENTIALS_MANDATORY)
+
+
+def test_update_when_data_has_modified_attributes(resource, mock_ansible_module):
+    data_merged = DICT_DEFAULT_STORAGE_SYSTEM.copy()
+    data_merged['credentials']['newIp_hostname'] = '10.10.10.10'
+
+    resource.get_by_ip_hostname.return_value = DICT_DEFAULT_STORAGE_SYSTEM
+    resource.update.return_value = data_merged
+
+    mock_ansible_module.params = yaml.load(YAML_STORAGE_SYSTEM_CHANGES)
+
+    StorageSystemModule().run()
+
+    mock_ansible_module.exit_json.assert_called_once_with(
+        changed=True,
+        msg=StorageSystemModule.MSG_UPDATED,
+        ansible_facts=dict(storage_system=data_merged)
+    )
+
+
+def test_update_when_data_has_modified_attributes_when_api500(resource, mock_ansible_module, mock_ov_client):
+    mock_ov_client.api_version = 500
+    data_merged = DICT_DEFAULT_STORAGE_SYSTEM_500.copy()
+    data_merged['credentials']['newHostname'] = '10.10.10.10'
+
+    resource.get_by_hostname.return_value = DICT_DEFAULT_STORAGE_SYSTEM_500
+    resource.update.return_value = data_merged
+
+    mock_ansible_module.params = yaml.load(YAML_STORAGE_SYSTEM_CHANGES_500)
+
+    StorageSystemModule().run()
+
+    mock_ansible_module.exit_json.assert_called_once_with(
+        changed=True,
+        msg=StorageSystemModule.MSG_UPDATED,
+        ansible_facts=dict(storage_system=data_merged)
+    )
+
+
+def test_should_remove_storage_system(resource, mock_ansible_module):
+    resource.get_by_ip_hostname.return_value = DICT_DEFAULT_STORAGE_SYSTEM
+
+    mock_ansible_module.params = yaml.load(YAML_STORAGE_SYSTEM_ABSENT)
+
+    StorageSystemModule().run()
+
+    mock_ansible_module.exit_json.assert_called_once_with(
+        changed=True,
+        msg=StorageSystemModule.MSG_DELETED
+    )
+
+
+def test_should_do_nothing_when_storage_system_not_exist(resource, mock_ansible_module):
+    resource.get_by_ip_hostname.return_value = []
+
+    mock_ansible_module.params = yaml.load(YAML_STORAGE_SYSTEM_ABSENT)
+
+    StorageSystemModule().run()
+
+    mock_ansible_module.exit_json.assert_called_once_with(
+        changed=False,
+        msg=StorageSystemModule.MSG_ALREADY_ABSENT
+    )
```

Comment:
Looks to be missing the `if __name__ == '__main__':` part to be able to run through `python <name_of_file>`